echo_handlers.go

```diff
@@ -25,337 +22,230 @@ import (
-	templates["alby/index.html"] = template.Must(template.ParseFS(embeddedViews, "views/backends/alby/index.html", "views/layout.html"))
-	templates["about.html"] = template.Must(template.ParseFS(embeddedViews, "views/about.html", "views/layout.html"))
-	templates["404.html"] = template.Must(template.ParseFS(embeddedViews, "views/404.html", "views/layout.html"))
-	templates["lnd/index.html"] = template.Must(template.ParseFS(embeddedViews, "views/backends/lnd/index.html", "views/layout.html"))
-	e.Renderer = &TemplateRegistry{
-		templates: templates,
-	}
 	e.HideBanner = true
 	e.Use(echologrus.Middleware())
-
 	e.Use(middleware.Recover())
 	e.Use(middleware.RequestID())
 	e.Use(middleware.CSRFWithConfig(middleware.CSRFConfig{
-		TokenLookup: "form:_csrf",
+		TokenLookup: "header:X-CSRF-Token",
 	}))
 	e.Use(session.Middleware(sessions.NewCookieStore([]byte(svc.cfg.CookieSecret))))
 	e.Use(ddEcho.Middleware(ddEcho.WithServiceName("nostr-wallet-connect")))
 
-	assetSubdir, _ := fs.Sub(embeddedAssets, "public")
-	assetHandler := http.FileServer(http.FS(assetSubdir))
-	e.GET("/public/*", echo.WrapHandler(http.StripPrefix("/public/", assetHandler)))
-	e.GET("/apps", svc.AppsListHandler)
-	e.GET("/apps/new", svc.AppsNewHandler)
-	e.GET("/apps/:pubkey", svc.AppsShowHandler)
-	e.POST("/apps", svc.AppsCreateHandler)
-	e.POST("/apps/delete/:pubkey", svc.AppsDeleteHandler)
-	e.GET("/logout", svc.LogoutHandler)
-	e.GET("/about", svc.AboutHandler)
-	e.GET("/", svc.IndexHandler)
+	e.GET("/api/csrf", svc.CSRFHandler)
+	e.GET("/api/info", svc.InfoHandler)
+	e.GET("/api/user/me", svc.UserMeHandler)
+	e.GET("/api/apps", svc.AppsListHandler)
+	e.GET("/api/apps/:pubkey", svc.AppsShowHandler)
+	e.DELETE("/api/apps/:pubkey", svc.AppsDeleteHandler)
+	e.POST("/api/apps", svc.AppsCreateHandler)
+	e.POST("/api/logout", svc.LogoutHandler)
+
+	frontend.RegisterHandlers(e)
 }
 
-func (svc *Service) IndexHandler(c echo.Context) error {
-	sess, _ := session.Get(CookieName, c)
-	returnTo := sess.Values["return_to"]
+func (svc *Service) CSRFHandler(c echo.Context) error {
+	csrf, _ := c.Get(middleware.DefaultCSRFConfig.ContextKey).(string)
+	return c.JSON(http.StatusOK, csrf)
+}
+
+func (svc *Service) InfoHandler(c echo.Context) error {
+	responseBody := &api.InfoResponse{}
+	responseBody.BackendType = svc.cfg.LNBackendType
+	return c.JSON(http.StatusOK, responseBody)
+}
+
+func (svc *Service) UserMeHandler(c echo.Context) error {
 	user, err := svc.GetUser(c)
 	if err != nil {
-		return err
-	}
-	if user != nil && returnTo != nil {
-		delete(sess.Values, "return_to")
-		sess.Options.MaxAge = 0
-		sess.Options.SameSite = http.SameSiteLaxMode
-		if svc.cfg.CookieDomain != "" {
-			sess.Options.Domain = svc.cfg.CookieDomain
-		}
-		sess.Save(c.Request(), c.Response())
-		return c.Redirect(302, fmt.Sprintf("%s", returnTo))
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Error:   true,
```

Comment:
why `Error: true`? is there a more standard way of returning error responses, and can we use a helper so it's consistent?